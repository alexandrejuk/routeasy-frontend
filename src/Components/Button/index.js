import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './style.module.css'

const Button = ({
  children,
  onClick,
  type,
}) => (
  <button
    className={classNames(styles.button, styles[type])}
    onClick={onClick}
    type="button"
  >
    {children}
  </button>
)

Button.propTypes = {
  children: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  type: PropTypes.oneOf([
    'danger',
    'primary',
  ]).isRequired,
}

export default Button
