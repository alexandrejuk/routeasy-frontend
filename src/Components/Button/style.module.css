@import '../../assets/css/colors.css';

.button {
  border: 1px solid var(--mine-shaft);
  border-radius: 1px;
  color: var(--white);
  cursor: pointer;
  font-size: 0.9em;
  font-weight: 600;
  padding: 8px 22px;
  text-align: center;
  transition: all 0.3s ease-in-out;
  width: 100%;
}

.primary {
  background-color: var(--silver-tree);
  color: var(--white);
}

.primary:hover,
.danger:hover {
  opacity: 0.9;
}

.danger {
  background-color: var(--pomegranate);
  color: var(--white);
}
