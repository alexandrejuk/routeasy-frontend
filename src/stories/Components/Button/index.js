import React from 'react'
import { action } from '@storybook/addon-actions'
import { Button } from '../../../Components/'

export default {
  title: 'Components/Button',
  component: Button,
}

const Template = ({
  children,
  onClick,
  type,
}) => (
  <Button
    onClick={onClick}
    type={type}
  >
    {children}
  </Button>
)

export const Primary = Template.bind({})

Primary.args = {
  children: 'Primary Button',
  onClick: action('onClick!'),
  type: 'primary',
}

export const Danger = Template.bind({})

Danger.args = {
  children: 'Danger Button',
  onClick: action('onClick!'),
  type: 'danger',
}
