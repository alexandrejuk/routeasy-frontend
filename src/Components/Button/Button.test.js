import React from 'react'
import {
  fireEvent,
  render,
} from '@testing-library/react'
import Button from './'

const PRIMARY_BUTTON_TEXT = 'Primary Button'
const DANGER_BUTTON_TEXT = 'Danger Button'

describe('Button component', () => {
  it('should render correctly primary button', () => {
    const callback = jest.fn()
    const { container, getByText } = render(
      <Button onClick={callback} type="primary" >
        {PRIMARY_BUTTON_TEXT}
      </Button>
    )
    const SampleButton = getByText(PRIMARY_BUTTON_TEXT)

    expect(callback).not.toHaveBeenCalled()

    fireEvent.click(SampleButton)

    expect(callback).toHaveBeenCalled()
    expect(container).toMatchSnapshot()
  })

  it('should render correctly danger button', () => {
    const callback = jest.fn()
    const { container, getByText } = render(
      <Button onClick={callback} type="danger">
        {DANGER_BUTTON_TEXT}
      </Button>
    )
    const SampleButton = getByText(DANGER_BUTTON_TEXT)

    expect(callback).not.toHaveBeenCalled()

    fireEvent.click(SampleButton)

    expect(callback).toHaveBeenCalled()
    expect(container).toMatchSnapshot()
  })
})
