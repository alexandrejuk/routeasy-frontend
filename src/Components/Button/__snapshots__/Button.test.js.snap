// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly danger button 1`] = `
<div>
  <button
    class="button danger"
    type="button"
  >
    Danger Button
  </button>
</div>
`;

exports[`Button component should render correctly primary button 1`] = `
<div>
  <button
    class="button primary"
    type="button"
  >
    Primary Button
  </button>
</div>
`;
